File: app/src/main/java/cn/com/woong/readhub/ui/widget/newsview/NewsAdapter.java
package cn.com.woong.readhub.ui.widget.newsview;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import java.util.ArrayList;

import javax.inject.Inject;

import cn.com.woong.readhub.App;
import cn.com.woong.readhub.R;
import cn.com.woong.readhub.bean.NewsMo;
import cn.com.woong.readhub.db.DBManager;

/**
 * Created by wong on 2018/3/9.
 */

public class NewsAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {

    private Context mContext;
    private ArrayList<NewsMo> mNewsMos = new ArrayList<>();
    private boolean mShowDelete;

//    @Inject
    public NewsAdapter(Context context) {
        mContext = context;
    }

    public void showDelete(boolean show) {
        mShowDelete = show;
    }

    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(App.Companion.getAppContext()).inflate(R.layout.item_news, parent, false);
        return new NewsViewHolder(mContext, view);
    }

    @Override
    public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {
        if (mShowDelete) {
            ((NewsViewHolder) holder).showDelete();
        }

        ((NewsViewHolder) holder).bind(mNewsMos.get(position));
    }

    @Override
    public int getItemCount() {
        return mNewsMos.size();
    }

    public void updateNews(boolean clear, ArrayList<NewsMo> newsMos) {
        if (clear) {
            mNewsMos.clear();
        }

        mNewsMos.addAll(newsMos);
        notifyDataSetChanged();
    }

    public void removeNews(int position) {
        DBManager.getInstance(mContext).deleteNewsMo(mNewsMos.get(position));
        mNewsMos.remove(position);
        notifyItemRemoved(position);
    }
}

File: app/src/main/java/cn/com/woong/readhub/ui/widget/newsview/NewsAdapter.kt
package cn.com.woong.readhub.ui.widget.newsview

import android.content.Context
import cn.com.woong.readhub.db.DBManager
import cn.com.woong.readhub.bean.NewsMo
import android.support.v7.widget.RecyclerView
import cn.com.woong.readhub.App
import android.view.LayoutInflater
import android.view.ViewGroup
import cn.com.woong.readhub.R


class NewsAdapter constructor(private val mContext: Context) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
    private val mNewsMos = arrayListOf<NewsMo>()
    private var mShowDelete: Boolean = false

    fun showDelete(show: Boolean) {
        mShowDelete = show
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        val view = LayoutInflater.from(App.sContext).inflate(R.layout.item_news, parent, false)
        return NewsViewHolder(mContext, view)
    }

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
        if (mShowDelete) {
            (holder as NewsViewHolder).showDelete()
        }

        (holder as NewsViewHolder).bind(mNewsMos.get(position))
    }

    override fun getItemCount(): Int {
        return mNewsMos.size
    }

    fun updateNews(clear: Boolean, newsMos: ArrayList<NewsMo>) {
        if (clear) {
            mNewsMos.clear()
        }

        mNewsMos.addAll(newsMos)
        notifyDataSetChanged()
    }

    fun removeNews(position: Int) {
        DBManager.getInstance(mContext).deleteNewsMo(mNewsMos[position])
        mNewsMos.removeAt(position)
        notifyItemRemoved(position)
    }
}
